util/logger.h

```diff
@@ -32,6 +32,9 @@ class Logger {
     writer_(writer)
   {}
 
+  Logger (const Logger&) = delete;
+  Logger& operator= (const Logger&) = delete;
+
```

Comment:
Because print_troubleshooting_message_and_exit() is now called from locations with and without a Logger the Logger arg was made std::optional which doesn't support references, and I noticed that it was silently making a copy of the Logger which seemed bad, so make Logger non-copyable by deleting copy ctor and copy assignment operator.